hoomd/md/data/local_access.py

```diff
@@ -22,10 +22,13 @@ def _cpp_cls(self):
         'virial': 'getVirial'
     }
 
-    def __init__(self, force_obj):
+    def __init__(self, force_obj, state):
         super().__init__()
         self._force_obj = force_obj
-        self._cpp_obj = self._cpp_cls(force_obj._cpp_obj)
+        data_cls = getattr(state._cpp_sys_def, force_obj._cpp_data_method)()
+        self._cpp_obj = self._cpp_cls(force_obj._cpp_obj, data_cls.getN(),
```

Comment:
So `N`, `N_ghosts`, and `N_global` are now passed through on construction. Previously, these were dynamically determined in the `getBuffer` call. I assume this change is OK because the context manager cannot be reused again? For example, the user couldn't construct a context manager, then change the number of particles, the enter the context manager (and seg fault because `N` is no longer valid).

Reply:
Not without accessing multiple internal (underscored) attributes and essentially ignoring our API.